### PR TITLE
Added a embed style for all command.

### PR DESCRIPTION
**Pull request by BadEnd777**
## Description

Added a embed style for the [report](src/commands/moderators/report.ts ':include :type=code') command.

## Changes

- **Deleted**

```ts
await channel.send(
  [
    `Reported message: https://discord.com/channels/${interaction.guild.id}/${interaction.channelId}/${interaction.targetId} (reported ${messageReportCount} times)`,
    `Message author: ${member.user} (reported ${reporteeCount} times)`,
    `Reported by: ${interaction.user} (sent ${reporterCount} reports)`,
  ].join('\n'),
)
```

- **Added**

```ts
const messagelink = `https://discord.com/channels/${interaction.guild.id}/${interaction.channelId}/${interaction.targetId}`

// TODO: Add a button to delete the message and ban the user (Mod only)
const embed = {
  title: 'Reported message',
  url: messagelink,
  color: 0xff0000,
  thumbnail: {
    url: member.user.displayAvatarURL(),
  },
  fields: [
    { name: 'Message author', value: member.toString(), inline: true },
    { name: 'Reported by', value: interaction.user.toString(), inline: true },
    {
      name: 'Reported message',
      value: `\`\`\`${message.content}\`\`\`\n[Jump to message](${messagelink})`,
    },
  ],
  timestamp: new Date(),
}

await channel.send({ embeds: [Object.assign(embed)] })
```

### CODE REVIEW

**Comment by buskun** [comment]
Can we separate the embed template creation to another function accepting required information?

`const getMessageReportingEmbed = (...) => ({ /* Embed object here */ })`

**Comment by BadEnd777** [comment]
yap, it can be done
but normally, we create a new embed for each command because the data we need is not the same
but there are some that can be used as a template, such as
image command that will only have an image and description

Ex: template of image command

```ts
cosnt ImageEmbed = (url: string, description: string) => ({
  title: 'Image',
  description,
  image: { url },
})
```

Ex: usage

```ts
await channel.send({
  embeds: [ImageEmbed("https://i.imgur.com/...", "This is a cat")],
});
```

**Comment by dtinth** [comment]
please add some screenshot so we see the end result krubb 🙏

**Comment by dtinth** [comment]
meanwhile, i just merged #36 because the implementation is simple and PR is very small scope and the PR author also provided a screenshot. but there are other valid improvements in this PR, please rebase krub 🙏

**Comment by buskun** [comment]
> yap, it can be done but normally, we create a new embed for each command because the data we need is not the same but there are some that can be used as a template, such as image command that will only have an image and description
> 
> Ex: template of image command
> 
> ```ts
> cosnt ImageEmbed = (url: string, description: string) => ({
>   description,
>   image: { url },
> })
> ```
> 
> Ex: usage
> 
> ```ts
> await channel.send({
>   embeds: [ImageEmbed("https://i.imgur.com/...", "This is a cat")],
> });
> ```
> 
> but actually the `discord.js` lib already has `EmbedBuilder` to use, but I didn't use it because I think it's not cool XD.

My purpose is not for reusability for this matter, it is more toward readability and testing (in the future).
Also, I am fine with both builder and raw one, but if we are using raw js object one, please also specify type for the embed variable.

File: README.md [hunk 1]
```diff
@@ -8,7 +8,15 @@ Discord bot for KaoGeek, built with TypeScript
 # Install dependencies
 pnpm install
 
+# Copy .env.example to .env
 cp .env.example .env
+
+# Change node version to 16
+nvm install v16 # if you don't have node 16 installed
```
Comment (beam41) [inline README.md:15]: project actually use node 18, you might want to change this

Comment (buskun) [reply README.md:15]: Actually, in [package.json](https://github.com/creatorsgarten/kaogeek-discord-bot/blob/cc9df638a75059b5f2983a3392ac1dab3a89071c/package.json#L42), it specifies both v16 and v18.

Comment (BadEnd777) [reply README.md:15]: Yap v16 and v18, but i'm use v19 is default.
I and someone must change node version to develop.

Thank you

**Comment by BadEnd777** [comment]
> please add some screenshot so we see the end result krubb 🙏

image

**Comment by BadEnd777** [comment]
> > yap, it can be done but normally, we create a new embed for each command because the data we need is not the same but there are some that can be used as a template, such as image command that will only have an image and description
> > Ex: template of image command
> > ```ts
> > cosnt ImageEmbed = (url: string, description: string) => ({
> >   description,
> >   image: { url },
> > })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Ex: usage
> > ```ts
> > await channel.send({
> >   embeds: [ImageEmbed("https://i.imgur.com/...", "This is a cat")],
> > });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but actually the `discord.js` lib already has `EmbedBuilder` to use, but I didn't use it because I think it's not cool XD.
> 
> My purpose is not for reusability for this matter, it is more toward readability and testing (in the future). Also, I am fine with both builder and raw one, but if we are using raw js object one, please also specify type for the embed variable.

I will try. Thank you.

File: src/client.ts [hunk 2]
```diff
@@ -38,16 +38,12 @@ export default class Bot extends Client {
...
-      cwd: this.__dirname,
-      root: this.__dirname,
-      absolute: true,
-    })
+    const eventFiles = globSync(
+      path.resolve(this.__dirname, 'events/*.{js,ts}'),
+    )
+    const commandFolders = globSync(
+      path.resolve(this.__dirname, 'commands/**/*.{js,ts}'),
+    )
```
Comment (beam41) [inline src/client.ts:46]: This revert #33 please change back

Comment (BadEnd777) [reply src/client.ts]: Back then.

File: README.md (hunk 1 above)
Comment (Leomotors) [reply README.md:15]: Change to `nvm install`. It will auto detect the version in .nvmrc

**Review by Leomotors — COMMENTED** [review]
Please run `pnpm format`

**Comment by BadEnd777** [comment]
> Please run `pnpm format`

all done. maybe.

**Pull request closed without merging**